runtime/src/pcr.rs

```diff
@@ -0,0 +1,124 @@
+// Licensed under the Apache-2.0 license
+
+use crate::Drivers;
+use caliptra_common::mailbox_api::{
+    ExtendPcrReq, IncrementPcrResetCounterReq, MailboxResp, MailboxRespHeader, QuotePcrsReq,
+    QuotePcrsResp,
+};
+use caliptra_drivers::{
+    cprintln, hand_off::DataStore, CaliptraError, CaliptraResult, Ecc384PrivKeyIn, KeyReadArgs,
+    PcrBank, PcrId,
+};
+use zerocopy::{transmute, FromBytes};
+
+pub struct PcrResetCounter {
+    counter: [u32; 32],
+}
+
+impl Default for PcrResetCounter {
+    fn default() -> Self {
+        PcrResetCounter::new()
+    }
+}
+
+impl PcrResetCounter {
+    fn new() -> PcrResetCounter {
+        PcrResetCounter { counter: [0; 32] }
+    }
+
+    pub fn get(&self, id: PcrId) -> u32 {
+        self.counter[usize::from(id)]
+    }
+
+    /// Increment the selected reset counter.
+    /// Returns `false` in case of a counter overflow
+    pub fn increment(&mut self, id: PcrId) -> bool {
+        let old_value = self.counter[usize::from(id)];
+
+        if let Some(new_value) = old_value.checked_add(1) {
+            self.counter[usize::from(id)] = new_value;
+
+            true
+        } else {
+            false
+        }
+    }
+}
+
+pub struct IncrementPcrResetCounter;
```

Comment:
nit: rename this IncrementPcrResetCounterCmd to maintain consistency with other mbox cmds?